dev/PipsPager/PipsPager.cpp

```diff
@@ -521,6 +528,49 @@ void PipsPager::OnNextButtonClicked(const IInspectable& sender, const winrt::Rou
     SelectedPageIndex(SelectedPageIndex() + 1);
 }
 
+void PipsPager::OnPipsAreaGettingFocus(const IInspectable& sender, const winrt::GettingFocusEventArgs& args)
+{
+    if (const auto repeater = m_pipsPagerRepeater.get())
+    {
+        // Easiest way to check if focus change came from within:
+        // Check if element is child of repeater by getting index and checking for -1
+        // If it is -1, focus came from outside and we want to get to selected element.
+        if (const auto oldFocusedElement = args.OldFocusedElement().try_as<winrt::UIElement>())
+        {
+            if (repeater.GetElementIndex(oldFocusedElement) == -1)
+            {
+                if (const auto realizedElement = repeater.GetOrCreateElement(SelectedPageIndex()).try_as<winrt::UIElement>())
+                {
+                    if (const auto argsAsIGettingFocusEventArgs2 = args.try_as<winrt::IGettingFocusEventArgs2>())
+                    {
+                        if (argsAsIGettingFocusEventArgs2.TrySetNewFocusedElement(realizedElement))
+                        {
+                            args.Handled(true);
+                        }
+                    }
+                    else
+                    {
+                        // Without TrySetNewFocusedElement, we cannot set focus while it is changing.
+                        m_dispatcherHelper.RunAsync([realizedElement]()
+                            {
+                                SetFocus(realizedElement, winrt::FocusState::Programmatic);
+                            });
+                        args.Handled(true);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void PipsPager::OnPipsAreaBringIntoViewRequested(const IInspectable& sender, const winrt::BringIntoViewRequestedEventArgs& args)
+{
+    if (Orientation() == winrt::Orientation::Vertical && isnan(args.VerticalAlignmentRatio()) || Orientation() == winrt::Orientation::Horizontal && isnan(args.HorizontalAlignmentRatio()))
```

Comment:
I guess you want to do is:
```
 if ((Orientation() == winrt::Orientation::Vertical && isnan(args.VerticalAlignmentRatio())) || 
(Orientation() == winrt::Orientation::Horizontal && isnan(args.HorizontalAlignmentRatio())))
```

Reply:
yes, you're right, thanks for catching that!